adafruit_esp32spi/adafruit_esp32spi_server.py

```diff
@@ -0,0 +1,193 @@
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+
+"""
+`adafruit_esp32spi_server`
+================================================================================
+
+TODO: better description?
+Server management lib to make handling and responding to incoming requests much easier
+
+* Author(s): Matt Costi
+"""
+# pylint: disable=no-name-in-module
+
+import os
+from micropython import const
+import adafruit_esp32spi.adafruit_esp32spi_socket as socket
+from adafruit_esp32spi.adafruit_esp32spi_requests import parse_headers
+
+_the_interface = None   # pylint: disable=invalid-name
+def set_interface(iface):
+    """Helper to set the global internet interface"""
+    global _the_interface   # pylint: disable=global-statement, invalid-name
+    _the_interface = iface
+    socket.set_interface(iface)
+
+NO_SOCK_AVAIL = const(255)
+INDEX_HTML = "/index.html"
+
+
+# pylint: disable=unused-argument, redefined-builtin, invalid-name
+class server:
+    """ TODO: class docs """
+    def __init__(self, port=80, debug=False):
+        self.port = port
+        self._server_sock = socket.socket(socknum=NO_SOCK_AVAIL)
+        self._client_sock = socket.socket(socknum=NO_SOCK_AVAIL)
+        self._debug = debug
+        self._listeners = {}
+        self._static_dir = None
+        self._static_files = []
+
+
+    def start(self):
+        """ start the server """
+        self._server_sock = socket.socket()
+        _the_interface.start_server(self.port, self._server_sock.socknum)
+        if self._debug:
+            ip = _the_interface.pretty_ip(_the_interface.ip_address)
+            print("Server available at {0}:{1}".format(ip, self.port))
+            print("Sever status: ", _the_interface.get_server_state(self._server_sock.socknum))
+
+    def on(self, method, path, request_handler):
+        """
+        Register a Request Handler for a particular HTTP method and path.
+        request_handler will be called whenever a matching HTTP request is received.
+
+        request_handler should accept the following args:
+            (Dict headers, bytes body, Socket client)
+
+        :param str method: the method of the HTTP request
+        :param str path: the path of the HTTP request
+        :param func request_handler: the function to call
+        """
+        self._listeners[self._get_listener_key(method, path)] = request_handler
+
+    def set_static_dir(self, directory_path):
```

Comment:
Again, we try to avoid `get` and `set` method names. This could be a property:
```python
@property
def static_dir(self):
    return self._static_dir

@static_dir.setter
def static_dir(self, directory_path):
   # current 'set_static_dir'
```